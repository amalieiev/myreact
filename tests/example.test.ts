// src/__tests__/example.js
// query utilities:
import {
    getByLabelText,
    getByText,
    getByTestId,
    queryByTestId,
    queries,
    waitFor,
    queryByText,
} from "@testing-library/dom";
// adds special assertions like toHaveTextContent
import "@testing-library/jest-dom";

test("examples of some things", async () => {
    const container = document.createElement("div");
    container.innerHTML = "<span>Print Username</span>";

    setTimeout(() => {
        container.innerHTML = "<button>Success</button>";
    }, 400);

    expect(container).toMatchSnapshot();

    expect(getByText(container, /Print Username/)).toBeTruthy();

    await waitFor(() => expect(queryByText(container, /Success/)).toBeTruthy());

    expect(container).toMatchSnapshot();
});
