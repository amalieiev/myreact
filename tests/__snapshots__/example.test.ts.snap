// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<div>
  <span>
    Print Username
  </span>
</div>
`;

exports[`examples of some things 2`] = `
<div>
  <button>
    Success
  </button>
</div>
`;
